### PR TITLE
chore!: remove HTTPServerConfig from the public api

### PR DESCRIPTION
**Pull request by mabdinur**
`ddtrace.config.http_server` should be used to set http server configurations.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by P403n1x87 — DISMISSED** [review]

**Review by brettlangdon — COMMENTED** [review]
I wonder if we should wait for the rebase/merge from `master` with #3163, and then the only change is removing the already deprecated `HTTPServerConfig`.

Otherwise we are going to have a merge conflict here.

**Comment by mabdinur** [comment]
> I wonder if we should wait for the rebase/merge from `master` with #3163, and then the only change is removing the already deprecated `HTTPServerConfig`.
> 
> Otherwise we are going to have a merge conflict here.

Yup, my plan is to merge all the deprecation warnings into master first (probably in batches of 2-3 PRs), rebase 1.0-dev and then merge the 1.0-dev changes

**Review by brettlangdon — CHANGES_REQUESTED** [review]
Since rebasing on `master` there are no changes in this PR anymore.

File: releasenotes/notes/internalize_http_server-06721b67e63dde1d.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    HTTPServerConfig was removed from the public api
```
Comment (brettlangdon) [inline releasenotes/notes/internalize_http_server-06721b67e63dde1d.yaml:4]:
```suggestion
    ``ddtrace.settings.HTTPServerConfig`` was removed from the public api.
```

**Review by brettlangdon — COMMENTED** [review]
last nit, then lgtm

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/internalize_http_server-06721b67e63dde1d.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    ``ddtrace.settings.HTTPServerConfig`` was removed from the public api.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/internalize_http_server-06721b67e63dde1d.yaml:4]:
```suggestion
    ``ddtrace.settings.Config.HTTPServerConfig`` was removed from the public api.
```

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**